webui/react/src/components/UserAvatar.tsx

```diff
@@ -3,19 +3,36 @@ import React from 'react';
 import Avatar, { Props as AvatarProps } from 'shared/components/Avatar';
 import useUI from 'shared/contexts/stores/UI';
 import { useUsers } from 'stores/users';
+import { DetailedUser } from 'types';
 import { Loadable } from 'utils/loadable';
 import { getDisplayName } from 'utils/user';
 
+import css from './UserAvatar.module.scss';
 export interface Props extends Omit<AvatarProps, 'darkLight' | 'displayName'> {
+  type?: 'table' | 'compactTable' | 'avatarOnly';
+  user?: DetailedUser;
   userId?: number;
 }
 
-const UserAvatar: React.FC<Props> = ({ userId, ...rest }) => {
+const UserAvatar: React.FC<Props> = ({ userId, user, type = 'avatarOnly', ...rest }) => {
   const users = Loadable.getOrElse([], useUsers()); // TODO: handle loading state
   const { ui } = useUI();
-  const displayName = getDisplayName(users.find((user) => user.id === userId));
+  const u = user ? user : users.find((user) => user.id === userId);
+  const displayName = getDisplayName(u);
 
-  return <Avatar {...rest} darkLight={ui.darkLight} displayName={displayName} />;
+  const avatar = <Avatar {...rest} darkLight={ui.darkLight} displayName={displayName} />;
+  if (type === 'avatarOnly' || !u) return avatar;
```

Comment:
Instead of having a branch here, this should really be a new component. UserLabel or something.